src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -191,197 +371,190 @@ private ExecutorInfo createExecutor(String taskIdName, String nodeName, String e
-        .setFailoverTimeout(conf.getFailoverTimeout())
-        .setUser(conf.getHdfsUser())
-        .setRole(conf.getHdfsRole())
-        .setCheckpoint(true);
+  private void reloadConfigsOnAllRunningTasks(SchedulerDriver driver) {
+    for (Protos.TaskStatus taskStatus : liveState.getRunningTasks().values()) {
+      sendMessageTo(driver, taskStatus.getTaskId(), taskStatus.getSlaveId(),
+          HDFSConstants.RELOAD_CONFIG);
+    }
+  }
 
-    try {
-      FrameworkID frameworkID = persistentState.getFrameworkID();
-      if (frameworkID != null) {
-        frameworkInfo.setId(frameworkID);
+  private void correctCurrentPhase() {
+    if (liveState.getJournalNodeSize() < conf.getJournalNodeCount()) {
+      liveState.transitionTo(AcquisitionPhase.JOURNAL_NODES);
+    } else if (liveState.getNameNodeSize() < HDFSConstants.TOTAL_NAME_NODES) {
+      liveState.transitionTo(AcquisitionPhase.START_NAME_NODES);
+    } else if (liveState.getNameNodeSize() == HDFSConstants.TOTAL_NAME_NODES) {
+      if (liveState.isNameNode1Initialized()
+          && liveState.isNameNode2Initialized()) {
+        liveState.transitionTo(AcquisitionPhase.DATA_NODES);
+      } else {
+        liveState.transitionTo(AcquisitionPhase.FORMAT_NAME_NODES);
       }
-    } catch (InterruptedException | ExecutionException | InvalidProtocolBufferException e) {
-      throw new RuntimeException(e);
+    } else {
+      log.error("Framework is in an unstable state, attention is required");
     }
-
-    MesosSchedulerDriver driver = new MesosSchedulerDriver(this, frameworkInfo.build(),
-        conf.getMesosMasterUri());
-    driver.run().getValueDescriptor().getFullName();
   }
-    
-  private void sendMessageTo(SchedulerDriver driver, TaskID taskId, String message) {
-    log.info(String.format("Sending message '%s' to taskId=%s", message, taskId.getValue()));
-    String slaveId = liveState.getTaskSlaveMap().get(taskId);
-    String postfix = taskId.getValue();
-    postfix = postfix.substring(postfix.indexOf(".") + 1, postfix.length());
-    postfix = postfix.substring(postfix.indexOf(".") + 1, postfix.length());
-    driver.sendFrameworkMessage(
-        ExecutorID.newBuilder().setValue("executor." + postfix).build(),
-        SlaveID.newBuilder().setValue(slaveId).build(),
-        message.getBytes());
+
+  private boolean offerNotEnoughResources(Offer offer, double cpus, int mem) {
+    for (Resource offerResource : offer.getResourcesList()) {
+      if (offerResource.getName().equals("cpus") &&
+          cpus > offerResource.getScalar().getValue()) {
+        return true;
+      }
+      if (offerResource.getName().equals("mem") &&
+          mem > offerResource.getScalar().getValue()) {
+        return true;
+      }
+    }
+    return false;
   }
 
+  private void reconcilePersistentState() {
+    LinkedHashMap<Protos.TaskID, Protos.TaskStatus> runningTasks = liveState.getRunningTasks();
+    for (String taskId : persistentState.getAllTaskIds()) {
```

Comment:
Running tasks are added and removed consistently in the persistent state list. Reconciliation happens upon reregistration to verify consistency when we reregister.